### ISSUE TITLE
Duplicate File awsSdkSigV4AConfigSelector.d.ts breaking deployments

### ISSUE THREAD

**Issue by noynek** [aws/aws-sdk-js-v3#6464]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

Our builds recently started failing when updating from 3.645.0 to 3.649.0, and we were able to trace it to duplicate `awsSdkSigV4AConfigSelector.d.ts` files. When I try to unzip our build zip file, I get this error. I'm guessing there's a case sensitivity issue.

### SDK version number

@aws-sdk/package-name@version, ...

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

v20.12.1

### Reproduction Steps

We're using azure with github actions so it's a little hard to provide repro instructions. Hoping the bug description is enough.

### Observed Behavior

Our logs showed
>Run From Package Initialization failed. Zip Contains 2 duplicate files with identical name in the same path
It was only when trying to unzip that we could see that the issue came from the aws package.

### Expected Behavior

zip-based builds to succeed. No duplicate files.

**Comment by timtucker-dte** [comment]
We also started seeing this yesterday.

Content in the 2 files is exactly the same as well.

**Comment by trivikr** [comment]
I'm not seeing the duplicate files in `@aws-sdk/core@3.649.0`

```console
$ mkdir test-install && cd test-install

$ test-install> npm install @aws-sdk/core@3.649.0 --save-exact

$ test-install> ls -1a node_modules/@aws-sdk/core/dist-es/submodules/httpAuthSchemes/aws_sdk    
.
..
AwsSdkSigV4ASigner.js
AwsSdkSigV4Signer.js
awsSdkSigV4AConfigSelector.js
index.js
resolveAwsSdkSigV4AConfig.js
resolveAwsSdkSigV4Config.j
```

**Comment by trivikr** [comment]
I'm able to see duplicate files in `@aws-sdk/core@3.649.0` in a case sensitive operating system like Linux

```console
$ mkdir test-install && cd test-install

$ test-install> npm install @aws-sdk/core@3.649.0 --save-exact

$ test-install> ls -1a node_modules/@aws-sdk/core/dist-es/submodules/httpAuthSchemes/aws_sdk
.
..
awsSdkSigV4AConfigSelector.js
AwsSdkSigV4AConfigSelector.js
AwsSdkSigV4ASigner.js
AwsSdkSigV4Signer.js
index.js
resolveAwsSdkSigV4AConfig.js
resolveAwsSdkSigV4Config.js
```

Also verified in npmjs file browser https://www.npmjs.com/package/@aws-sdk/core?activeTab=code

**Comment by trivikr** [comment]
This happened as TurboRepo cached artifact from an intermediate state of the `@aws-sdk/core` from a workspace, and it reused that cache from release since the source artifacts hadn't changed.

A quick-fix would be clean like `rimraf dist-es && build:es` to all our package.json build scripts.

**Comment by noynek** [comment]
Thank you very much.

**Comment by trivikr** [comment]
The fix is merged in https://github.com/aws/aws-sdk-js-v3/pull/6467, and new version will be published in https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.652.0 around 12:30 PM Pacific on Friday Sep 13th.

**Comment by trivikr** [comment]
> new version will be published in https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.651.1

Verified that issue is fixed in `v3.651.1`

```console
$ test-install> npm install @aws-sdk/core@3.651.1 --save-exact

$ test-install> ls -1a node_modules/@aws-sdk/core/dist-es/submodules/httpAuthSchemes/aws_sdk
.
..
AwsSdkSigV4ASigner.js
AwsSdkSigV4Signer.js
index.js
resolveAwsSdkSigV4AConfig.js
resolveAwsSdkSigV4Config.js
```
